shared/eip712/eip712.go

```diff
@@ -0,0 +1,54 @@
+package eip712
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+)
+
+type EIP712Components struct {
+	R [32]byte `json:"r"`
+	S [32]byte `json:"s"`
+	V uint8    `json:"v"`
+}
+
+const EIP712Length = 65
+
+// Pretty print for EIP712Components
+func (e EIP712Components) String() {
+	fmt.Println("EIP712 Components:")
+	fmt.Printf("R: %x\n", e.R)
+	fmt.Printf("S: %x\n", e.S)
+	fmt.Printf("V: %d\n", e.V)
+}
+
+// Decode decodes a hex-encoded EIP-712 signature string, verifies the length
+// and assigns the appropriate bytes to R/S/V
+func (e *EIP712Components) Decode(inp string) error {
```

Comment:
mea culpa, let's rename Decode to UnmarshalText and accept inp as []byte
and rename Encode to MarshalText and return ([]byte, error)

so we can fulfill this interface: <https://pkg.go.dev/encoding#TextMarshaler>